models/Image.php

```diff
@@ -175,35 +210,56 @@ public function ensureThumbnailPath()
      */
     public function generateThumbnail($crop = null)
     {
-        ImagineImage::$thumbnailBackgroundAlpha = 0;
         $size = Yii::$app->params['image.size.thumbnail'];
-        
-        if ($crop === null) {
-            ImagineImage::thumbnail($this->getOriginalPath(), $size[0], $size[1], ImageInterface::THUMBNAIL_INSET)
-                ->save($this->ensureThumbnailPath());    
-        } else {
-            ImagineImage::crop($this->getOriginalPath(), $crop['width'], $crop['height'], [$crop['x'], $crop['y']])
-                ->thumbnail(new Box($size[0], $size[1]), ImageInterface::THUMBNAIL_INSET)
-                ->save($this->ensureThumbnailPath());
+
+        $image = new SimpleImage($this->getOriginalPath());
```

Comment:
It provides convenient API, especially for https://github.com/claviska/SimpleImage/blob/master/src/claviska/SimpleImage.php#L506